### PR TITLE
expose knowhere config as go package

### PR DESCRIPTION
**Pull request by Presburger**
/kind improvement
issue: #757

### CODE REVIEW

**Comment by Presburger** [comment]
> @Presburger may I ask why this is needed? p.s. the code looks good

@alexanderguzhva   Thanks, We want to unify and reuse the config check logic of the Milvus proxy frontend with the checking code in Knowhere. Currently, some of the checks in the Golang frontend are completely independent, which can lead to inconsistencies between the two sides.

**Comment by Presburger** [comment]
issue: #757

**Review by foxspy — APPROVED** [review]
/lgtm

File: knowhere.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import "C"
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"unsafe"
+)
+
+func CheckConfig(name string, params map[string]string, ptype string) error {
```
Comment (liliu-z) [inline knowhere.go:17]: Suggest to put the cgo function in segcore aligning with all other APIs.
We can put the `IndexType - number` mapping in Knowhere side.

**Pull request closed without merging**